src/NuGet.Services.AzureSearch/SearchService/AzureSearchQueryBuilder.cs

```diff
@@ -38,171 +38,175 @@ internal class AzureSearchQueryBuilder
+            }
+            else
+            {
+                _result.Append('(');
+
+                for (var i = 0; i < terms.Count; i++)
                 {
-                    // We are building the query's first clause, only add the required operator "+" if there are other top-level clauses.
-                    // We generate a top-level clause for each non-field-scoped term and one for each field-scopes.
-                    if (_nonFieldScopedValues.Count > 0 || _fieldScopedValues.Keys.Count > 1)
+                    if (i > 0)
                     {
-                        result.Append('+');
+                        _result.Append(' ');
                     }
-                }
-                else
-                {
-                    // We are adding another top-level clause to the query, always add the required operator "+".
-                    result.Append(" +");
-                }
 
-                result.Append(fieldScopedTerm.Key);
-                result.Append(':');
-
-                if (fieldScopedTerm.Value.Count == 1)
-                {
-                    AppendEscapedString(result, fieldScopedTerm.Value[0]);
-                }
-                else
-                {
-                    result.Append('(');
-                    AppendEscapedValues(result, fieldScopedTerm.Value);
-                    result.Append(')');
+                    AppendScopedTerm(terms[i], prefixSearch);
                 }
+
+                _result.Append(')');
             }
+        }
 
-            if (_nonFieldScopedValues.Any())
-            {
-                if (result.Length > 0)
-                {
-                    result.Append(' ');
-                }
+        private void AppendScopedTerm(string term, bool prefixSearch)
+        {
+            AppendEscapedString(term.Trim(), quoteWhiteSpace: !prefixSearch);
 
-                AppendEscapedValues(result, _nonFieldScopedValues);
+            if (prefixSearch)
+            {
+                _result.Append('*');
             }
-
-            return result.ToString();
         }
 
-        private static void AppendEscapedValues(StringBuilder result, IReadOnlyList<string> values)
+        private void AppendEscapedValues(IReadOnlyList<string> values, bool prefixSearch)
         {
             for (var i = 0; i < values.Count; i++)
             {
                 if (i > 0)
                 {
-                    result.Append(' ');
+                    _result.Append(' ');
                 }
 
-                AppendEscapedString(result, values[i]);
+                AppendEscapedString(values[i], quoteWhiteSpace: !prefixSearch);
+
+                if (prefixSearch)
+                {
+                    _result.Append('*');
+                }
             }
         }
 
-        private static void AppendEscapedString(StringBuilder result, string input)
+        private void AppendEscapedString(string input, bool quoteWhiteSpace = true)
```

Comment:
is it necessary have default parameters here? IMO it improves GitHub readability to avoid this and name parameters
(nit)

Reply:
Made parameters required